src/lib/coda_base/ledger_hash.ml

```diff
@@ -34,24 +34,37 @@ module Merkle_tree =
 
 let depth = Coda_compile_config.ledger_depth
 
-include Data_hash.Make_full_size ()
-
-module Arg = struct
-  type t = Stable.Latest.t [@@deriving bin_io_unversioned]
-
+include Data_hash.Make_full_size (struct
   let description = "Ledger hash"
 
   let version_byte = Base58_check.Version_bytes.ledger_hash
-end
+end)
+
+(* Data hash versioned boilerplate below *)
+
+[%%versioned
+module Stable = struct
+  module V1 = struct
+    module T = struct
+      type t = Field.t [@@deriving sexp, compare, hash, version {asserted}]
+    end
+
+    include T
+
+    let to_latest = Core.Fn.id
+
+    let to_yojson = to_yojson
```

Comment:
think you can use the newfangled `[%%define_from_scope (to_yojson,of_yojson)]`

Reply:
or not, more useful when the list is long.

Reply:
aha, I was looking at define_locally and saw a TODO about it. glad to know this exists!